src/store/ViewStore.js

```diff
@@ -46,13 +48,29 @@ export default class ViewStore {
     }
 
     @action.bound saveViews() {
-        this.updateRoute('main');
-        if (this.templateName.length > 0) {
+        if (this.views.some(x=>x.name.toLowerCase() === this.templateName.toLowerCase())){
+            this.dialog.setOpen(true , false);
+        }
+        else if (this.templateName.length > 0) {
+            this.updateRoute('main');
             const layout = this.stx.exportLayout();
             this.views.push({ name: this.templateName, layout });
             this.updateLocalStorage();
+            this.templateName = '';
+        }
+    }
+
+    @action.bound onOverwrite() {
+        if (this.templateName.length > 0) {
```

Comment:
it's odd that we need this. User should not be able to change the input of the template or add anymore templates until they have dealt with the prompt.